Cairo Desktop/CairoDesktop.Common/XmlConfigStackLocation.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Xml.Serialization;
+using ManagedShell.Common.Common;
+using ManagedShell.Common.Helpers;
+using ManagedShell.ShellFolders;
+
+namespace CairoDesktop.Common
+{
+    public class XmlConfigStackLocation : StackLocation
+    {
+        private readonly XmlSerializer _serializer;
+        private readonly ThreadSafeObservableCollection<ShellFile> _files;
+        private readonly FileSystemWatcher _watcher;
+
+        private string _configFile;
+        private string _displayName;
+
+        public XmlConfigStackLocation(string configFile)
+        {
+            _serializer = System.Xml.Serialization.XmlSerializer.FromTypes(new[] { typeof(List<string>) })[0];
+            _configFile = configFile;
+            _displayName = System.IO.Path.GetFileNameWithoutExtension(_configFile);
+            _files = new ThreadSafeObservableCollection<ShellFile>();
+
+            LoadConfig();
+
+            _watcher = new FileSystemWatcher();
+            _watcher.Path = System.IO.Path.GetDirectoryName(_configFile);
+            _watcher.Filter = System.IO.Path.GetFileName(_configFile);
+            _watcher.EnableRaisingEvents = true;
+            _watcher.Changed += _watcher_Changed;
+            _watcher.Renamed += _watcher_Renamed;
+        }
+        
+        private void _watcher_Renamed(object sender, RenamedEventArgs e)
+        {
+            _watcher.EnableRaisingEvents = false;
+
+            _configFile = e.FullPath;
+            _displayName = System.IO.Path.GetFileNameWithoutExtension(_configFile);
+
+            LoadConfig();
+            
+            _watcher.Path = System.IO.Path.GetDirectoryName(_configFile);
+            _watcher.Filter = System.IO.Path.GetFileName(_configFile);
+            _watcher.EnableRaisingEvents = true;
+        }
+
+        private void LoadConfig()
+        {
+            _files.Clear();
+            var shellFiles = Load();
+            foreach (var path in shellFiles)
+            {
+                ShellFile shellFile = new ShellFile(null, path);
```

Comment:
`parentFolder` should always be non-null, since context menus are loaded by the parent folder (they don't work with this commit). 2 thoughts:
- Maybe ManagedShell should enforce this?
- Maybe the parent could be the desktop folder?

Reply:
It looks like context menus on non-xml locations are also broken, but I'm not sure why.